lib/bap/bap_self.ml

```diff
@@ -60,4 +60,66 @@ module Create() = struct
   let warning f = message Warning ~section:name f
   let error f = message Error ~section:name f
 
+  module Config = struct
+    let plugin_name = name
+    include Bap_config
+
+    let (/) = Filename.concat
+    let confdir = confdir / plugin_name
+    let conf_filename = confdir / "config"
+
+    let get_env_options () =
+      let prefix = "BAP_" ^ String.uppercase plugin_name ^ "_" in
+      let prefix_chop_key (k, v) =
+        match String.chop_prefix ~prefix k with
+        | Some k -> Some (k, v)
+        | None -> None in
+      let plugin_filter_map str =
+        match String.split str ~on:'=' with
+        | k :: vs -> prefix_chop_key (k, String.concat ~sep:"=" vs)
+        | [] -> None in
+      Unix.environment () |>
+      Array.to_list |>
+      List.filter_map ~f:plugin_filter_map
+
+    let conf_file_options () =
+      let string_splitter str =
+        match String.split str ~on:'=' with
+        | k :: vs -> k, String.concat ~sep:"=" vs
+        | [] -> invalid_arg "empty string" in
+      let split_filter = List.map ~f:string_splitter in
+      try
+        In_channel.with_file
+          conf_filename ~f:(fun ch -> In_channel.input_lines ch
+                                      |> split_filter)
+      with Sys_error _ -> []
+
+    let options () =
+      get_env_options () |>
+      List.fold ~init:(conf_file_options ())
+        ~f:(fun o (k, v) -> List.Assoc.add o k v)
+
+    let get = options () |> List.Assoc.find ~equal:String.Caseless.equal
```

Comment:
In such a scenario (i.e. by implementing it like this), `options ()` cannot return all the bindings.

Also, this _is_ cross platform since, as far as I could tell from reading online, both Unix-style OSs (Linux and Mac) and Windows would return it as a list of strings, with the strings having `=` as the separator.